src/tsickle.ts

```diff
@@ -166,18 +166,19 @@ class ClosureRewriter extends Rewriter {
     // then paramTags[0] = [info about x, info about y].
     const paramTags: jsdoc.Tag[][] = [];
     const returnTags: jsdoc.Tag[] = [];
+    const typeParameterNames = new Set<string>();
```

Comment:
Imagine you have three overloads of your function, they might be using the same template type parameter multiple times. I've added a test for that.